app/lib/admin/actions/resolve_moderation_case.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:clock/clock.dart';
+import 'package:pub_dev/admin/models.dart';
+import 'package:pub_dev/shared/datastore.dart';
+
+import 'actions.dart';
+
+final resolveModerationCase = AdminAction(
+  name: 'resolve-moderation-case',
+  summary: 'Closes the moderation case and resolves a final status.',
+  description: '''
+Closes the moderation case and updates the status based on the actions logged on the case.
+''',
+  options: {
+    'case': 'The caseId to be closed.',
```

Comment:
I think we should consider a `--resolution` argument as well, we can make it optional and then infer by default.